src/lib/containers/TotalQueryResults.tsx

```diff
@@ -11,72 +13,52 @@ export type TotalQueryResultsProps = {
   frontText: string
 }
 
-type State = {
-  total: number
-  isLoading: boolean
-}
-
 // This is a stateful component so that during load the component can hold onto the previous
 // total instead of showing 0 results for the intermittent loading state.
-export default class TotalQueryResults extends React.Component<
-  TotalQueryResultsProps,
-  State
-> {
-  constructor(props: TotalQueryResultsProps) {
-    super(props)
-    this.state = {
-      total: 0,
-      isLoading: false,
-    }
-  }
-
-  componentDidMount() {
-    this.calculateTotal()
-  }
 
-  componentDidUpdate(prevProps: TotalQueryResultsProps) {
-    // Check that it has entered a loading state, meaning that there is new data
-    if (this.props.isLoading && !prevProps.isLoading) {
-      this.calculateTotal()
-    }
-  }
-
-  calculateTotal() {
-    const { getLastQueryRequest, token } = this.props
-    const queryRequest = getLastQueryRequest!()
-    queryRequest.partMask = SynapseConstants.BUNDLE_MASK_QUERY_COUNT
+const TotalQueryResults: React.FunctionComponent<TotalQueryResultsProps> = ({
+  style,
+  unitDescription,
+  frontText,
+  getLastQueryRequest,
+  token,
+  isLoading: parentLoading,
+}) => {
+  const [total, setTotal] = React.useState(0)
+  const [isLoading, setIsLoading] = React.useState(false)
 
-    this.setState({
-      isLoading: true,
-    })
-    SynapseClient.getQueryTableResults(queryRequest, token)
-      .then(data => {
-        this.setState({
-          total: data.queryCount!,
-          isLoading: false,
+  React.useEffect(() => {
+    const calculateTotal = async () => {
+      const queryRequest = getLastQueryRequest!()
+      queryRequest.partMask =
+        SynapseConstants.BUNDLE_MASK_QUERY_COUNT |
+        SynapseConstants.BUNDLE_MASK_QUERY_FACETS
+      setIsLoading(true)
+      SynapseClient.getQueryTableResults(queryRequest, token)
+        .then(data => {
+          setTotal(data.queryCount!)
+        })
+        .catch(err => {
+          console.error('err ', err)
         })
-      })
-      .catch(err => {
-        console.error('err ', err)
-        this.setState({
-          isLoading: false,
+        .finally(() => {
+          setIsLoading(false)
         })
-      })
-  }
+    }
+    calculateTotal()
+  }, [parentLoading, token, getLastQueryRequest])
```

Comment:
I think this useEffect would calculate the facet results twice for every time a query is run, once when the query is kicked off (parenLoading = true) and then once more after its finished (parentLoading = false), but I believe that it should only calculate the query once (parentLoading = true). I think the logic to check parentLoading has changed to true need's to be tracked and checked, I have this logic in my PR [here](https://github.com/Sage-Bionetworks/Synapse-React-Client/pull/464/files#diff-838d52273514f635c18c31b169632988R92) but for token change.

I believe each time this component gets new props getLastQueryRequest will be a new function thats passed in, although it would (or could) contain the same query request from before. I think one way around this could be to pass in the last query request directly and use useDeepCompareEffect on that object, [like this](https://github.com/Sage-Bionetworks/Synapse-React-Client/blob/305e181dc8c57ee9563d0dd899dd5144c875c3e9/src/lib/containers/download_list/DownloadConfirmation.tsx#L144).

Reply:
My intuition would be to kick it off once parentLoading == false (when things settle). The performance is indeed better when it gets kicked off while parent is loading ( parentLoading == true). And it does make sense since the query they are executing is the same. I did have to modify the test for it to pass. But I just want to bring it up to make sure that kicking it off while parent is still loading does not produce any unwanted effects

Reply:
I agree with that assessment! I guess there's a fairly small possibility of a race condition introduced here - if the user makes two query updates quickly then its possible that the second result is overriden by the first result if the two come back out of sync.

e.g.

1. Make facet selection (parentLoading === true), total query results kicks off a query, call it A
2. Make facet selection, (the following still holds parentLoading === true) so total query results kicks off a second query, call it B
3. B resolves and setTotal is called
4. A resolves and setTotal is called
5. TotalQueryResults now has an incorrect total from A instead of B

I dunno how much this is worth worrying about, it _seems_ pretty unlikely

Reply:
I still have concern with getLastQueryRequest being passed as a dependency, I'm mostly concerned that it _could_ be a problem, its an unknown.

Reply:
May be I'm missing something but in practical terms when would the props change? We have :
```
 style,
  unitDescription,
  frontText,
  getLastQueryRequest,
  token,
  isLoading: parentLoading,
```
seems like only last 3 are the ones that would get changed during the life on the component and we do need to watch for those any way.  On a related note, why do we not pass lastQueryRequest directly to the QueryWraper child components and pass the function instead?

Reply:
That's a good point about passing lastQueryRequest directly, I think you're right we should do that! then useDeepCompareEffect can get be used to do the deep diff on the object. And I agree that only the last 3 should change change, seems right to me too.

Reply:
I have observed that when we are fetching when `parentLoading === true` - the first time the table comes up after clicking on left nav the total is 0. This might be the case where the the case of race condition or something else so I have changed it `parentLoading === false` check